sqlglot/dialects/duckdb.py

```diff
@@ -146,6 +147,44 @@ def _build_greatest(args: t.List) -> exp.Greatest:
     return exp.Greatest(this=seq_get(args, 0), expressions=args[1:])
 
 
+def _approx_quantiles_sql(self: DuckDB.Generator, expression: exp.ApproxQuantiles) -> str:
+    """
+    BigQuery's APPROX_QUANTILES(expr, n) returns an array of n+1 approximate quantile values
+    dividing the input distribution into n equal-sized buckets.
+
+    Both BigQuery and DuckDB use approximate algorithms for quantile estimation, but BigQuery
+    does not document the specific algorithm used so results may differ. DuckDB does not
+    support RESPECT NULLS.
+    """
+    this = expression.this
+    if isinstance(this, exp.Distinct):
+        # APPROX_QUANTILES requires 2 args and DISTINCT node grabs both
+        if len(this.expressions) < 2:
+            self.unsupported("APPROX_QUANTILES requires a bucket count argument")
+            return self.function_fallback_sql(expression)
+        num_quantiles_expr = this.expressions.pop(1)
```

Comment:
Nit: small defensive rewrite to ensure the AST does not end up in an inconsistent state after this `pop`.

```suggestion
        num_quantiles_expr = this.expressions[1].pop()
```